firebase-devservices/deployment/src/test/java/io/quarkiverse/googlecloudservices/firebase/deployment/testcontainers/FirebaseEmulatorContainerIntegrationTest.java

```diff
@@ -0,0 +1,359 @@
+package io.quarkiverse.googlecloudservices.firebase.deployment.testcontainers;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import com.google.api.core.ApiFuture;
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.api.gax.grpc.GrpcTransportChannel;
+import com.google.api.gax.rpc.FixedTransportChannelProvider;
+import com.google.cloud.NoCredentials;
+import com.google.cloud.firestore.*;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.cloud.pubsub.v1.TopicAdminClient;
+import com.google.cloud.pubsub.v1.TopicAdminSettings;
+import com.google.cloud.storage.*;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseAuthException;
+import com.google.firebase.auth.UserRecord;
+import com.google.firebase.database.*;
+import com.google.firebase.internal.EmulatorCredentials;
+import com.google.firebase.internal.FirebaseProcessEnvironment;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.PubsubMessage;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+
+@Testcontainers
+public class FirebaseEmulatorContainerIntegrationTest {
+
+    private static final File tempEmulatorDataDir;
+    private static final File tempHostingContentDir;
+
+    static {
+        try {
+            // Create a temporary directory for emulator data
+            tempEmulatorDataDir = Files.createTempDirectory("firebase-emulator-data").toFile();
+            tempHostingContentDir = Files.createTempDirectory("firebase-hosting-content").toFile();
```

Comment:
again, please create files somewhere it will be automatically deleted

Reply:
on it

Reply:
Fixed https://github.com/quarkiverse/quarkus-google-cloud-services/pull/707/commits/338451a8c93df85c097d2ad005e9ccdf653c0a3a